pkg/proxy/userspace/proxier.go

```diff
@@ -578,6 +586,7 @@ func (proxier *Proxier) unmergeService(service *v1.Service, existingPorts sets.S
 			klog.Error(err)
 		}
 		proxier.loadBalancer.DeleteService(serviceName)
+		info.finished.Done()
```

Comment:
similar question about being called more than once (sending the counter negative)

Reply:
@liggitt in this case, cleanupPortalAndProxy() -> stopProxy() removes the ServiceInfo from the map and the waitgroup being called is a member of that ServiceInfo, so it cannot be called again. I thought about putting the finished closer to the map deletion, but we want to cover the loadbalancer cleanup too.